File: server/src/protocol/SnapshotCommand.cpp
#include "SnapshotCommand.h"
#include "CommandVisitor.h"

using namespace std;
using namespace communication;

SnapshotCommand::SnapshotCommand()
        : Command(10, CommandType::SNAPSHOT)
{}

SnapshotCommand::~SnapshotCommand()
{}

vector<uint8_t> SnapshotCommand::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(commandType_));

    return frame;
}

std::string SnapshotCommand::getName()
{
    return string("SnapshotCommand");
}

void SnapshotCommand::accept(CommandVisitor &visitor)
{
    visitor.visit(*this);
}

void SnapshotCommand::initializeDataSize()
{
    uint16_t dataSize = sizeof(commandType_);

    setDataSize(dataSize);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/SnapshotCommand.java
package wpam.mobile_client.protocol;
import java.util.ArrayList;

public final class SnapshotCommand extends Command
	{
		public SnapshotCommand()
		{
			super(10, CommandType.SNAPSHOT);
		}
		public final void close()
		{
			super.close();
		}

		@Override
		public ArrayList<Integer> getFrameBytes()
		{
			initializeDataSize();

			ArrayList<Integer> frame = getHeader();
			frame.add(commandType_.getValue());

			return frame;
		}
		@Override
		public String getName()
		{
			return new String("SnapshotCommand");
		}

		@Override
		public void accept(CommandVisitor visitor)
		{
			visitor.visit(this);
		}

		@Override
		protected void initializeDataSize()
		{
			setDataSize(1);
		}
	}